src/sdam/srv_polling.ts

```diff
@@ -114,7 +137,11 @@ export class SrvPoller extends TypedEventEmitter<SrvPollerEvents> {
   success(srvRecords: dns.SrvRecord[]): void {
     this.haMode = false;
     this.schedule();
-    this.emit(SrvPoller.SRV_RECORD_DISCOVERY, new SrvPollingEvent(srvRecords));
+    const event = new SrvPollingEvent(srvRecords);
+    if (!this.lastSrvPollingEvent?.equals(event)) {
+      this.emit(SrvPoller.SRV_RECORD_DISCOVERY, event);
+    }
```

Comment:
@dariakp 
This is my test of a current discussion around if hosts should be "sticky" or each srv polling event should cause a reshuffling. 
I POC-ed this and didn't mean to commit it whoops! I'll revert this but keep the tests fixes I made on your current comments